src/main/java/com/snowgears/grapplinghook/GrapplingHook.java

```diff
@@ -13,186 +16,183 @@
-							Bukkit.getPlayer(args[1]).getInventory().addItem(HookAPI.createGrapplingHook(50));
-							Bukkit.getPlayer(args[1]).sendMessage(ChatColor.GRAY+player.getName()+" has given you a grappling hook with 50 uses!");
-						}
-						else
-							player.sendMessage(ChatColor.RED+"Incorrect arguments. '/gh give <player>'.");
-					}
-					else
-						player.sendMessage(ChatColor.DARK_RED+"You are not authorized to do that.");
-				}
-			}
-			else{ //sender from console
-				if ((cmd.getName().equalsIgnoreCase("gh") && args[0].equalsIgnoreCase("give") && args[1].length() > 0)) {
-						if(Bukkit.getPlayer(args[1]) != null){
-							Bukkit.getPlayer(args[1]).getInventory().addItem(HookAPI.createGrapplingHook(50));
-							Bukkit.getPlayer(args[1]).sendMessage(ChatColor.GRAY+" You have been given a grappling hook with 50 uses by the server!");
-						}
-						else
-							sender.sendMessage(ChatColor.RED+"Incorrect arguments. '/gh give <player>'.");
-				}
-			}
-			return true;
-		}
-		else if(args.length == 3){
-			if(sender instanceof Player){
-				Player player = (Player)sender;
-				if (cmd.getName().equalsIgnoreCase("gh") && args[0].equalsIgnoreCase("give")) {
-					if(player.hasPermission("grapplinghook.command.give")){
-						if(isInteger(args[2])){
-							if(Bukkit.getPlayer(args[1]) != null){
-								int uses = Integer.parseInt(args[2]);
-								Bukkit.getPlayer(args[1]).getInventory().addItem(HookAPI.createGrapplingHook(uses));
-								Bukkit.getPlayer(args[1]).sendMessage(ChatColor.GRAY+player.getName()+" has given you a grappling hook with "+uses+" uses!");
-							}
-							else
-								player.sendMessage(ChatColor.RED+"That player could not be found. '/gh give <player> <#>'.");
-								
-						}
-						else
-							player.sendMessage(ChatColor.RED+"Incorrect arguments. '/gh give <player> <#>'.");
-					}
-					else
-						player.sendMessage(ChatColor.DARK_RED+"You are not authorized to do that.");
-				}
-			}
-			else{ //sending from console
-				if (cmd.getName().equalsIgnoreCase("gh") && args[0].equalsIgnoreCase("give")) {
-						if(isInteger(args[2])){
-							if(Bukkit.getPlayer(args[1]) != null){
-								int uses = Integer.parseInt(args[2]);
-								Bukkit.getPlayer(args[1]).getInventory().addItem(HookAPI.createGrapplingHook(uses));
-								Bukkit.getPlayer(args[1]).sendMessage(ChatColor.GRAY+"You have been given a grappling hook with "+uses+" uses by the server!");
-							}
-							else
-								sender.sendMessage(ChatColor.RED+"That player could not be found. '/gh give <player> <#>'.");
-								
-						}
-						else
-							sender.sendMessage(ChatColor.RED+"Incorrect arguments. '/gh give <player> <#>'.");
-				}
-			}
-			return true;
-		}
+        timeBetweenUses = getConfig().getInt("timeBetweenGrapples");
+
+        if (disableCrafting == false) {
+            for (Material plankMaterial : Tag.PLANKS.getValues()) {
+                ShapedRecipe woodRecipe = new ShapedRecipe(NamespacedKey.randomKey(),
```

Comment:
You should **never** use a random NamespacedKey.
The whole purpose of a NamespacedKey is for it to uniquely identify an object.
By passing a new random key everytime, you violate this very important property.

Besides, namespacedkeys are supposed to be created using the plugin instance...